docs/DeveloperGuide.adoc

```diff
@@ -77,6 +77,40 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+=== Use case: Rename tag
+
+*MSS*
+
+. User requests to list all tags
+. AddressBook shows a list of tags
+. User requests to edit a specific tag in the list
+. AddressBook requests user to type in the new tag name
+. User types in new tag name
+. AddressBook confirms with user again whether he/she wants to make the edit
+. User confirms edit
+. AddressBook edits tag name to the new tag name
```

Comment:
Maybe could mention that AddressBook finds and renames `all` existing instances of tag to be more specific